src/ddmd/backend/gflow.c

```diff
@@ -324,18 +324,44 @@ STATIC void accumrd(vec_t GEN,vec_t KILL,elem *n)
                 rdelem(&Gl,&Kl,n->E1);
                 rdelem(&Gr,&Kr,n->E2);
 
-                /* GEN = (GEN - Kl) | Gl |      */
-                /*      (GEN - Kr) | Gr         */
-                /* KILL |= Kl & Kr              */
+                switch (el_noreturn(n->E1) * 2 | el_noreturn(n->E2))
+                {
+                    case 0: // E1 and E2 return
+                        /* GEN = (GEN - Kl) | Gl |
+                         *       (GEN - Kr) | Gr
+                         * KILL |= Kl & Kr
+                         */
+                        vec_orass(Gl,Gr);
+                        vec_sub(Gr,GEN,Kl);
+                        vec_orass(Gl,Gr);
+                        vec_sub(Gr,GEN,Kr);
+                        vec_or(GEN,Gl,Gr);
+
+                        vec_andass(Kl,Kr);
+                        vec_orass(KILL,Kl);
```

Comment:
If you want to go for it (using `& ~` instead of `-` for clarity).
```d
(GEN & ~Kl) | (GEN & ~Kr) == (GEN & ~(Kl & Kr))
```
So you could do the `vec_andass(Kl,Kr);` that's needed anhow first, and then simplify the rest.
**Untested**
```d
vec_andass(Kl,Kr);
vec_orass(KILL,Kl);

vec_orass(Gl,Gr);
vec_sub(Gr,GEN,Kl); // (GEN - (Kl & Kr))
vec_or(GEN,Gl,Gr);
```